### PR TITLE
Dev refactor

### PR DESCRIPTION
**Pull request by EmiliaCiastek**
Communication between client and server changed

### CODE REVIEW

File: client/src/main/java/connection/chain/EndLink.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package connection.chain;
+
+import gui.playing.DispatcherAdapter;
+import responses.Response;
+
+public class EndLink implements Chain {
+
+  @Override
+  public void setNextChain(Chain nextChain) { //TODO: is it a good/bad practice?
```
Comment (patiem) [inline client/src/main/java/connection/chain/EndLink.java]: remove todo

File: client/src/test/java/connection/chain/ChainConfigFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+public class ChainConfigFactoryTest {
+
+  @Test
+  public void whenConfigureThenReturnsInstanceOfHitLink() {
+    // Act
+    Chain firstChain = ChainConfigFactory.configureChainOfResponsibilities();
+    // Assert
+    assertThat(firstChain).isInstanceOf(HitLink.class);
+  }
+}
```
Comment (patiem) [inline client/src/test/java/connection/chain/ChainConfigFactoryTest.java]: enter :p

File: client/src/test/java/connection/chain/EndLinkTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  @Test(expectedExceptions = UnsupportedOperationException.class)
+  public void whenSetNextChainThenThrowException() {
+    EndLink endLink = new EndLink();
+    Chain mockedNextChain = mock(Chain.class);
+
+    endLink.setNextChain(mockedNextChain);
+  }
+}
```
Comment (patiem) [inline client/src/test/java/connection/chain/EndLinkTest.java]: enter :p

File: client/src/test/java/connection/chain/HitLinkTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    when(mockedNoneResponse.getHeader()).thenReturn(ResponseHeader.NONE);
+
+    //Act
+    hitLink.analyzeResponse(mockedNoneResponse, dispatcherAdapter);
+
+    //Assert
+    verify(mockedNextChain, times(expectedInvocations))
+        .analyzeResponse(mockedNoneResponse, dispatcherAdapter);
+  }
+}
```
Comment (patiem) [inline client/src/test/java/connection/chain/HitLinkTest.java]: enter :p

File: client/src/test/java/connection/chain/LostLinkTest.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    //Act
+    lostLink.analyzeResponse(mockedNoneResponse, dispatcherAdapter);
+
+    //Assert
+    verify(mockedNextChain, times(expectedInvocations))
+        .analyzeResponse(mockedNoneResponse, dispatcherAdapter);
+  }
+
+}
```
Comment (patiem) [inline client/src/test/java/connection/chain/LostLinkTest.java]: enter :p

File: utils/src/main/java/json/JsonGeneratorAdapter.java [hunk 6]
```diff
@@ -2,14 +2,16 @@
...
  * @version 1.5
  */
 public class JsonGeneratorAdapter {
-  public String createJson(Object object, ObjectMapper objectMapper) throws JsonProcessingException {
-    return objectMapper.writeValueAsString(object);
+  public String createJson(Object object, ObjectMapper mapper) throws JsonProcessingException {
+    mapper.registerModule(new Jdk8Module());
+    return mapper.writeValueAsString(object);
   }
 }
```
Comment (patiem) [inline utils/src/main/java/json/JsonGeneratorAdapter.java]: enter :p

File: utils/src/main/java/responses/HitResponse.java [hunk 7]
```diff
@@ -2,13 +2,13 @@
...
+public class HitResponse extends Response {
   private ResponseHeader header = ResponseHeader.HIT;
 
-
   @JsonProperty("header")
   @Override
   public ResponseHeader getHeader() {
     return header;
   }
+
```
Comment (patiem) [inline utils/src/main/java/responses/HitResponse.java]: line

File: utils/src/main/java/responses/Response.java [hunk 8]
```diff
@@ -1,6 +1,32 @@
...
+  public abstract ResponseHeader getHeader();
+
+  @JsonProperty("shot")
+  public Optional<Shot> getShot() {
+    return Optional.empty();
+  }
 }
+
+
+
```
Comment (patiem) [inline utils/src/main/java/responses/Response.java]: many lines

File: server/src/main/java/communication/ConnectionHandler.java [hunk 9]
```diff
@@ -54,13 +55,7 @@ private PlayerClient createClient(final Socket socket) throws IOException {
...
   }
 
-  private void sendConnectionConfirmationMessage(PlayerClient playerClient) throws IOException {
-    final String playerIsConnected = "CON";
-    MessageSender messageSender = new MessageSender();
-    messageSender.sendMessageToPlayer(playerClient, playerIsConnected);
-  }
-
-  private ConnectionMessage prepareJSONMessage(BufferedReader reader) throws IOException {
+  private ConnectionMessage prepareJSONMessage(BufferedReader reader) throws IOException { //TODO: what this method is doing -> better name?
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/ConnectionHandler.java]: Solve TODO,

File: server/src/main/java/communication/PlayerClient.java [hunk 10]
```diff
@@ -11,7 +11,7 @@
  * @author Bartosz Pieczara / Emilia Ciastek
  * @version 1.5
  */
-class PlayerClient {
+public class PlayerClient {
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/PlayerClient.java]: why changing to public?

File: server/src/main/java/communication/Server.java [hunk 11]
```diff
@@ -32,14 +32,10 @@ public void runServer() {
...
-  private int setUpServerConfig() throws IOException{
-      Properties properties = new Properties();
-      InputStream config = ClassLoader.getSystemResourceAsStream(fileName);
-      properties.load(config);
-      return Integer.parseInt(properties.getProperty("portNumber"));
+  private int setUpServerConfig() throws IOException {
+    Properties properties = new Properties();
+    InputStream config = ClassLoader.getSystemResourceAsStream(SERVER_CONFIG_FILE);
+    properties.load(config);
+    return Integer.parseInt(properties.getProperty("portNumber"));
```
Comment (szczepanskikrs) [inline server/src/main/java/communication/Server.java]: :+1:

File: server/src/main/java/engine/GameRunner.java [hunk 12]
```diff
@@ -31,59 +43,48 @@ public GameRunner(final Round round, final PlayerRegistry playerRegistry) {
...
-  private void sendMessage(final ShotResult result, Shot shot) {
-    String message = result.toString();
-
-    String messageToOpponent = makeMessageForOpponent(result, shot);
-
-    if (gameState == GameState.WIN) {
-      message = gameState.toString();
-      messageToOpponent = "LOST";
-    }
+  private void logShotInfo(final Shot shot, final ShotResult shotResult) {
```
Comment (szczepanskikrs) [inline server/src/main/java/engine/GameRunner.java]: :+1:

**Pull request merged**